src/ray/core_worker/reference_count.cc

```diff
@@ -166,6 +166,23 @@ void ReferenceCounter::AddOwnedObject(const ObjectID &object_id,
   }
 }
 
+void ReferenceCounter::RemoveOwnedObject(const ObjectID &object_id) {
+  auto it = object_id_refs_.find(object_id);
+  if (it == object_id_refs_.end()) {
+    RAY_LOG(WARNING) << "Tried to remove reference for nonexistent owned object "
+                     << object_id;
+    return;
+  }
+  if (it->second.RefCount() != 0) {
+    RAY_LOG(WARNING) << "Tried to remove reference for owned object " << object_id
+                     << " that has " << it->second.RefCount()
+                     << " references, must have 0 references to be removed";
+    return;
+  }
```

Comment:
This should return an error or just fatally error instead of failing silently given that this should never happen (it's an incorrect use of the ref counter API if it does)